amplitude_python_sdk/v2/client.py

```diff
@@ -81,3 +89,75 @@ def batch_upload_events(
         """  # pylint: disable=line-too-long
         req_data = self._construct_event_request_data(events, options)
         return self._log_events_internal(routes.BATCH_API, req_data, timeout)
+
+    @staticmethod
+    def _construct_chart_request_data(
+        annotation: ChartAnnotationsV2
+    ) -> Dict[str, Any]:
+        """
+        Convert ChartAnnotationsV2 model to dictionary payload for _chart_annotations_post request
+        """
+        req_data = annotation.dict(exclude_none=True)
+
+        req_data["date"] = req_data["date"].__str__()
```

Comment:
I think we can fix the date serialization slightly differently. Pydantic has an elegant solution to this through configurable JSON encoders: https://pydantic-docs.helpmanual.io/usage/exporting_models/#json_encoders, and then we can use the `.json()` function to just generate the JSON output directly.

Reply:
Also, in general you'd invoke the `__str__` method like:

```python
req_data["date"] = str(req_data["date"])
```

Reply:
Converting the model to a dictionary and passing into `data=` instead of `json=` works.